### DISCUSSION TITLE
6. Odev hakkında

### DISCUSSION THREAD

**Discussion by OmerK-Benkyzz** [gusanmaz/BMB212-Algorithms-2223#29] [Q&A] [answered]
Hocam Merhabalar,
Stack.java diye bir java classı oluşuturlunca javanın stack importunda sorun yaşıyor. Stacks.java diye bir class oluşturduğumda sorunsuz çalışıyor bir sorun oluşturur mu testiniz açısından.

**Comment by gusanmaz** [comment]
Sorun olmamasi lazim. `Stack.java` seklinde isimlendirmenizi istiyorum Stack sinifinizi.

**Reply by OmerK-Benkyzz** [reply]
sorun yaşanıyor çünkü javanın Stack  classıyla oluşturmak istediğimiz Stack classı aynı ada sahip referans sorunu yaşanıyor.importların ayrı bir şekilde kullanılması gerekiyor isterseniz mailden detaylıca sorunu gösterebilirim.

**Reply by gusanmaz** [reply]
`Stack.java` be `Main.java` ayni klasorde olursa Stack sinifini kullanabilmeniz icin import etmenize gerek kalmiyor. https://github.com/gusanmaz/BMB212-Algorithms-22-23/discussions/16

**Reply by gusanmaz** [reply, marked as answer]
Bu ipucunu aslinda hic vermemem lazim ama basit bir Stack yaziminin genel olarak bu kadar sorun yaratabilecegini, bu konuda pek cok soru gelebilecegini tahmin etmemistim.

Buradaki 14. sayfadaki Stack kodunu referans alabilirsiniz.
https://algs4.cs.princeton.edu/lectures/keynote/13StacksAndQueues.pdf
